client/src/components/AttendeeInput.tsx

```diff
@@ -30,41 +29,60 @@ export default function AttendeeInput({ id, onChange, value, type }: AttendeeInp
       }
     }
   };
+  const extractEmails = (input: string[]) => {
+    return input
+      .join(' ')
+      .split(/\s|,/)
+      .map((email) => email.trim())
+      .filter((email) => email !== '');
+  };
+
+  const validateEmails = (emails: string[]) => {
+    const validEmails = emails.filter(isEmailValid);
+    const invalidEmails = emails.filter((email) => !isEmailValid(email));
+    return { validEmails, invalidEmails };
+  };
+
+  const handleInvalidEmails = (invalidEmails: string[]) => {
+    invalidEmails.forEach((email: string) => {
+      toast.error(`${email} is invalid.`);
+    });
+  };
 
   const handleSelectionChange = (_: React.SyntheticEvent, newValue: Array<string | IPeopleInformation>) => {
     const emails = newValue.map((option) => (typeof option === 'object' && option.email ? option.email : (option as string)));
-    const filteredEmails = emails.filter((email) => emails.indexOf(email) === emails.lastIndexOf(email));
-    if (filteredEmails.length > 0) {
-      const lastValue = filteredEmails[filteredEmails.length - 1].trim();
-      if (isEmailValid(lastValue)) {
-        onChange(id, filteredEmails);
-        setTextInput('');
-      } else {
-        toast.error('Invalid email entered');
-      }
-    } else {
-      onChange(id, filteredEmails);
-      setTextInput('');
+    const filteredEmails = extractEmails(emails);
+    const { validEmails, invalidEmails } = validateEmails(filteredEmails);
+    if (invalidEmails.length > 0) {
+      handleInvalidEmails(invalidEmails);
     }
+    const uniqueEmails = [...new Set(validEmails)];
```

Comment:
this could be done at the beginning to reduce the total result set when checking invalid emails